privacyidea/static/components/audit/controllers/auditControllers.js

```diff
@@ -65,6 +65,11 @@ myApp.controller("auditController", function (AuditFactory, $scope, $rootScope,
             $scope.getParams();
             AuditFactory.get($scope.params, function(data) {
                 $scope.auditdata = data.result.value;
+                if ($scope.auditdata.auditdata[0] && (typeof $scope.audit_columns !== 'undefined')) {
```

Comment:
I am not sure, how javascript behaves here. If  ``$scope.auditdata.auditdata`` is not defined, we will get a javascript exception. Can this happen?

Why do you test for ``$scope.audit_colums``? I think this is not necessary.

Reply:
done, thank's

Reply:
an empty data.result.value is not probable in the webUI. It works if the request works properly. If you have no access, this code will not run. If no entries are returned, data.result.value is defined also.

The typeof thing is not necessary. I introduced it to have the if-block only run once in the ng-repeater, since there was a related error. But apparently it stemmed from somewhere else.

Changed.